Resources/Computing/ParallelLibraries/Srun.py

```diff
@@ -0,0 +1,106 @@
+To work, srun needs to be used in conjonction with Slurm.
+"""
+import os
+import shutil
+import commands
+import stat
+
+from DIRAC.Resources.Computing.ParallelLibraries.ParallelLibrary import ParallelLibrary
+
+
+class Srun(ParallelLibrary):
+
+  def __init__(self, workingDirectory):
+    super(Srun, self).__init__('srun', workingDirectory)
+
+  def generateWrapper(self, executableFile):
+    """
+    Associate the executable with srun, to execute the same command in parallel on multiple nodes.
+    Wrap it in a new executable file
+
+    :param str executableFile: name of the executable file to run
+    :return str: name of the wrapper that runs the executable via srun
+    """
+    # As the executable will be executed by srun, it needs GRP and OTH execution permission
+    if not os.access(executableFile, 5):
+      os.chmod(executableFile, stat.S_IRWXU | stat.S_IRGRP | stat.S_IXGRP | stat.S_IROTH | stat.S_IXOTH)
+
+    # Build the script to run the executable in parallel multiple times
+    # srun is the command to execute a task multiple time in parallel
+    # -l option: add the task ID to the output
+    # -k option: do not kill the slurm job if one of the nodes is broken
+    cmd = '#!/bin/bash\n'
+    cmd += 'srun -l -k %s' % executableFile
+    self.log.debug("Command generated by Srun:", "%s" % cmd)
+
+    srunWrapper = os.path.join(self.workingDirectory, 'srunExec.sh')
+    with open(srunWrapper, 'w') as exe:
+      exe.write("%s" % cmd)
+    if not os.access(srunWrapper, 5):
+      os.chmod(srunWrapper, stat.S_IRWXU | stat.S_IRGRP | stat.S_IXGRP | stat.S_IROTH + stat.S_IXOTH)
+    return srunWrapper
+
+  def processOutput(self, output, error, isFile=True):
+    """
+    Reorder the content of the output files according to the node identifier.
+
+    From:
+    >>> 1: line1
+    >>> 2: line1
+    >>> 1: line2
+    To:
+    >>> # On node 1
+    >>>   line1
+    >>>   line2
+    >>> # On node 2
+    >>>   line1
+
+    :param str output: name of the output file, or its content
+    :param str error: name of the error file, or its content
+    :param bool isFile: indicates if the inputs represent files or content of the files
+    """
+    if isFile:
+      self._openFileAndSortOutput(output)
+      self._openFileAndSortOutput(error)
+    else:
+      output = self._sortOutput(output)
+      error = self._sortOutput(error)
+    return (output, error)
+
+  def _openFileAndSortOutput(self, outputFile):
+    """
+    Open a file, get its content and reorder it according to the node identifiers
+
+    :param str outputFile: name of the file to sort
+    """
+    with open(outputFile, 'r') as f:
+      outputContent = f.readlines()
+
+    sortedContent = self._sortOutput(outputContent)
+
+    with open(outputFile, 'w') as f:
+      f.write("%s" % sortedContent)
+
+  def _sortOutput(self, outputContent):
+    """
+    Reorder the content of the output file according to the node identifiers
+
+    :param str outputContent: content to sort
```

Comment:
This claims `outputContent` is supposed to be a string but assumes its a list of lines

Reply:
There was an issue actually. `outputContent` is supposed to be a string containing multiple lines, I added a line of code to make a list of lines.